### PR TITLE
RANGER-4952: Add support for Hive 4.0.1

### PR DESCRIPTION
**Pull request by simhadri-g**
## What changes were proposed in this pull request?
Hive 4.0 major version has been release. This is a significant release with over 5000+ commits compared to hive-3.x

As a result, we would like to upgrade the version of hive used in ranger to Hive-4.0.1 .

## How was this patch tested?

Unit tests and manual tests

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -125,7 +125,7 @@
         <ozone.version>1.4.0</ozone.version>
         <hamcrest.version>2.2</hamcrest.version>
         <hbase.version>2.6.0</hbase.version>
-        <hive.version>3.1.3</hive.version>
+        <hive.version>4.0.0</hive.version>
```
Comment (ayushtkn) [inline pom.xml]: 4.0.1

Comment (simhadri-g) [reply pom.xml]: Done, i have changed it to 4.1

Comment (simhadri-g) [reply pom.xml]: Thanks for the review!

**Comment by simhadri-g** [comment]
@vperiasamy  can you please review the PR.

**Comment by zhangbutao** [comment]
+1 for the change.
I can contribute to add connector ranger auth such as https://issues.apache.org/jira/browse/HIVE-25214 in Hive4 if we can get the change in Ranger.

**Review by rameeshm — APPROVED** [review]

**Comment by simhadri-g** [comment]
@rameeshm can you please merge the change if everything is fine? 
Thanks!

**Comment by simhadri-g** [comment]
thanks !

**Pull request merged**